src/components/permissions/routes.ts

```diff
@@ -1,44 +1,81 @@
 import databaseConnect from '../databaseConnect'
-import { getUsername, getUserInfo } from './loginHelpers'
+import { getUserInfo, getTokenData, extractJWTfromHeader } from './loginHelpers'
 import { updateRowPolicies } from './rowLevelPolicyHelpers'
 import bcrypt from 'bcrypt'
+import { Organisation, User, UserOrg } from '../../types'
 
 const saltRounds = 10 // For bcrypt salting: 2^saltRounds = 1024
 
-const routeUserPermissions = async (request: any, reply: any) => {
-  const token = (request?.headers?.authorization || '').replace('Bearer ', '')
-  const username = await getUsername(token)
-  return reply.send(await getUserInfo(username))
+const routeCreateHash = async (request: any, reply: any) => {
+  const { password } = request.body
+
+  // bcrypt hash output includes salt and other metadata in string
+  // See https://github.com/kelektiv/node.bcrypt.js#hash-info
+  const hash = await bcrypt.hash(password, saltRounds)
+  return reply.send({
+    hash,
+  })
 }
 
+/*
+Authenticates login and returns:
+  - userInfo
+  - list of organisations belonging to user
+  - template permissions
+  - JWT containing permissions and userId
+*/
 const routeLogin = async (request: any, reply: any) => {
-  const { username, password } = request.body || { username: '', password: '' }
-  const { passwordHash } = (await databaseConnect.getUserDataByUsername(username)) || {}
+  const { username, password } = request.body
+  if (password === undefined) return reply.send({ success: false })
 
-  if (!passwordHash) return reply.send({ success: false })
+  const userOrgInfo: UserOrg[] = (await databaseConnect.getUserOrgData({ username })) || {}
 
-  if (!(await bcrypt.compare(password, passwordHash))) return reply.send({ success: false })
+  const { userId, firstName, lastName, email, dateOfBirth, passwordHash } = userOrgInfo[0]
```

Comment:
looks like only the userId and passowrdHash are used in this method

Reply:
> @CarlosNZ, while looking at #181, i ran the test, and i think it's failing due to changes to compileJWT interface

Okay, looking into now. Thx

Reply:
Okay, should all be passing now. (after a `yarn database_init`)